### ISSUE TITLE
skopeo copy --encryption-key to "dir:" : no candidate MIME types

### ISSUE THREAD

**Issue by ChengyuZhu6** [containers/skopeo#1971]
When use skopeo copy an image with encryption from docker registry to the format of local directory,  command:
```
$ openssl genrsa -out private.key 1024
$ openssl rsa -in private.key -pubout > public.key
$ skopeo copy --debug --encryption-key jwe:./public.key docker://library/busybox dir:busybox 
```
there is an error:
```
DEBU[0000] overwriting existing container image directory "/home/zcy/kbs/busybox" 
DEBU[0000] Using registries.d directory /etc/containers/registries.d 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf" 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf.d/shortnames.conf" 
DEBU[0000] Trying to access "docker.io/library/busybox:latest" 
DEBU[0000] No credentials matching docker.io/library/busybox found in /run/user/1000/containers/auth.json 
DEBU[0000] No credentials matching docker.io/library/busybox found in /root/.config/containers/auth.json 
DEBU[0000] No credentials matching docker.io/library/busybox found in /root/.docker/config.json 
DEBU[0000] No credentials matching docker.io/library/busybox found in /root/.dockercfg 
DEBU[0000] No credentials for docker.io/library/busybox found 
DEBU[0000]  No signature storage configuration found for docker.io/library/busybox:latest, using built-in default file:///var/lib/containers/sigstore 
DEBU[0000] Looking for TLS certificates and private keys in /etc/docker/certs.d/docker.io 
DEBU[0000] GET https://registry-1.docker.io/v2/         
DEBU[0001] Ping https://registry-1.docker.io/v2/ status 401 
DEBU[0001] GET https://auth.docker.io/token?scope=repository%3Alibrary%2Fbusybox%3Apull&service=registry.docker.io 
DEBU[0003] GET https://registry-1.docker.io/v2/library/busybox/manifests/latest 
DEBU[0005] Content-Type from manifest GET is "application/vnd.docker.distribution.manifest.list.v2+json" 
DEBU[0005] Using blob info cache at /var/lib/containers/cache/blob-info-cache-v1.boltdb 
DEBU[0005] Source is a manifest list; copying (only) instance sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5 for current system 
DEBU[0005] GET https://registry-1.docker.io/v2/library/busybox/manifests/sha256:acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5 
DEBU[0005] Content-Type from manifest GET is "application/vnd.docker.distribution.manifest.v2+json" 
DEBU[0005] IsRunningImageAllowed for image docker:docker.io/library/busybox:latest 
DEBU[0005]  Using default policy section                
DEBU[0005]  Requirement 0: allowed                      
DEBU[0005] Overall: allowed                             
Getting image source signatures
DEBU[0005] Reading /var/lib/containers/sigstore/library/busybox@sha256=acaddd9ed544f7baf3373064064a51250b14cfe3ec604d65765a53da5958e5f5/signature-1 
DEBU[0005] Not looking for sigstore attachments: disabled by configuration 
DEBU[0005] Manifest has MIME type application/vnd.docker.distribution.manifest.v2+json, ordered candidate list [] 
FATA[0005] copying system image from manifest list: Internal error: no candidate MIME types
```
my skopeo version:
```
skopeo version 1.11.1-dev commit: 2363dfeeabd53d5c4f1e244765116b1c611e2f29
```

**Comment by mtrmac** [comment]
Thanks for your report.

This is a bug/missing feature in the c/image library; the image is using schema2, and needs to be converted to OCI for encryption to be possible at all.

Short-term, a workaround is to use `skopeo copy --format oci …`; c/image should learn to do something like that automatically.

**Comment by mtrmac** [comment]
This was fixed in https://github.com/containers/image/pull/1930 .

**Comment by mtrmac** [comment]
… and included in Skopeo by #1995 .
